### PR TITLE
V0/sia/unit tests

### PR DESCRIPTION
**Pull request by ghost**
Added UnitTests for CustomerService

### CODE REVIEW

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs [hunk 1]
```diff
@@ -0,0 +1,239 @@
+using ITGuru.FourWheels.Data.Interfaces;
+using ITGuru.FourWheels.Data;
+using ITGuru.FourWheels.Service;
+using Microsoft.VisualStudio.TestPlatform.TestHost;
+using System.ComponentModel.DataAnnotations;
+using System.Xml;
+
+namespace ITGuru.FourWheels.Tests.RepositoryTests
+{
+    public class UnitTest1
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs]: Shouldn't it be called something more appropriate? 🤔

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+            Assert.Equal(_DEFAULT_CUSTOMER.Email, createdCustomer.Email);
+
+            // 2. Delete
+            // Act
+            var removeResult = customerRepository.Remove(createdCustomer);
+            var deletedCustomer = customerRepository.GetById(createdCustomer.Id);
+
+            // Assert
+            Assert.Null(deletedCustomer);
+            Assert.False(removeResult.Succeeded);
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs]: Shouldn't this be `Assert.True`?

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+            // Act
+            // Save customer in repository.
+            var addResult = customerRepository.Add(_DEFAULT_CUSTOMER);
+
+            // Retrieve the newly created customer from the repository.
+            var createdCustomer = customerRepository.GetById(_DEFAULT_CUSTOMER.Id);
+
+            // Assert
+            Assert.False(addResult.Succeeded);
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs]: Shouldn't this be `Assert.True`?

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+                    LastName = "Three",
+                    Phone = "33333333",
+                    Email = "test3@itguru.com"
+                }
+            };
+
+            bool addAllSuccess = true;
+            foreach (var customer in customersToCreate)
+            {
+                addAllSuccess = customerRepository.Add(customer).Succeeded && addAllSuccess;
```
Comment (ZhakalenDk) [inline ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs:164]: I like this line <3

**Review by ZhakalenDk — CHANGES_REQUESTED** [review]
Some minor changes are required 👌

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs (hunk 1 above)
Comment (ghost) [reply ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs]: Of course, that has been corrected.

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs (hunk 2 above)
Comment (ghost) [reply ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs]: Fixed

**Review by ZhakalenDk — APPROVED** [review]
Looks good! Make it happen!

File: ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs (hunk 3 above)
Comment (ZhakalenDk) [reply ITGuru.FourWheels/ITGuru.FourWheels.Tests/RepositoryTests/CustomerRepositoryTest.cs]: Got resolved wiht latest commit 👌

**Pull request merged**